pykern/pkcollections.py

```diff
@@ -176,6 +242,54 @@ class PKDictNameError(NameError):
     pass
 
 
+def canonicalize(obj):
+    """Convert to lists and PKDicts for simpler serialization
+
+    Traverse `obj` to convert all values to forms that are compatible
+    with serialization protocols like YAML or JSON.
+
+    Simple objects are ensured to match their types e.g. bool, float,
+    int, and str.  Objects that are instances of these, are converted
+    to these to ensure they are basic types, that is,
+    ``canonicalize(str_subclass('a'))`` will be conveted to ``str('a')``.
+
+    bytes and bytearrays will be converted to str.
+
+    decimal.Decimal will converted to float.
+
+    All objects are traversed. If no objects need to be converted,
+    `obj` will be returned unmodified.
+
+    Generators and iterables are converted to lists.
+
+    Circularities are not detected so infinite recursion can occur.
+
+    Args:
+        obj (object): what to convert
+
+    Returns:
+        object: converted object (may or may not be the same)
+    """
+    o = obj
+    if o is None:
+        return o
+    # Order matters so we don't convert bools to ints, since bools are ints.
+    for x in bool, int, float, str:
```

Comment:
What about this kind of thing instead of long list of if blocks?
```python
PKDict(
    bool=bool(o),
    int=int(o),
   ...
   collections.abc.Iterable=list(canonicalize(i) for i in iter(o))
)[type(o)]